### ISSUE TITLE
`STAKE_POOLS_LIST` tests fail due pools appearing out of order.

### ISSUE THREAD

**Issue by jonathanknowles** [cardano-foundation/cardano-wallet#1940]
# Context

As part of the stake pool integration tests, we verify that pools within the test cluster have the  retirement epochs that we expect.

However, this test relies on pools being listed in a certain order, which turns out to be a condition that we can't currently rely on.

# Test Case

`STAKE_POOLS_LIST_01 / pools have the correct retirement information`

# Failure / Counter-example

https://hydra.iohk.io/build/3564934

According to the [log](https://hydra.iohk.io/build/3564934/nixlog/1/tail), the following expectation failed:

https://github.com/input-output-hk/cardano-wallet/blob/2f2cf55296c3b6a988ecf267b1045c19fdce0404/lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs#L610-L611

The value that _actually_ appeared was:
```hs
Just (ApiEpochInfo {epochNumber = ApiT {getApiT = EpochNo {unEpochNo = 1000000}}, epochStartTime = 2021-10-28 04:39:02.004223 UTC})")
```

In other words, `1_000_000` appeared when `1_000` was expected.

These two values are defined here:
https://github.com/input-output-hk/cardano-wallet/blob/2f2cf55296c3b6a988ecf267b1045c19fdce0404/lib/shelley/test/integration/Main.hs#L181-L186

This appears to be a simple out-of-order issue.

# Suggested Resolution 

Make the test agnostic to the order in which stake pools appear.

# QA 

https://github.com/input-output-hk/cardano-wallet/pull/1941

**Comment by piotr-iohk** [comment]
ok
